File: CarRentalApp/CarRentalApp/ClientApp/src/app/models/order.ts
import { User } from './user';
import { Car } from './car';

export class Order {
    id: number;
    comment: string;
    rentalBeginDate: Date;
    rentalEndDate: Date;
    userId: number;
    carId: number;
    
    user: User;
    car: Car;
}

File: CarRentalApp/DAL/Models/Order.cs
using System;

namespace DAL.Models
{
    public class Order
    {
        public int Id { get; set; }
        public string Comment { get; set; }
        public DateTime RentalBeginDate { get; set; }
        public DateTime RentalEndDate { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }

        public int CarId { get; set; }
        public Car Car { get; set; }
    }
}
